backend/src/test/java/gov/cdc/usds/simplereport/service/UploadServiceTest.java

```diff
@@ -0,0 +1,71 @@
+package gov.cdc.usds.simplereport.service;
+
+import com.fasterxml.jackson.databind.RuntimeJsonMappingException;
+import gov.cdc.usds.simplereport.db.model.Person;
+import gov.cdc.usds.simplereport.db.model.StreetAddress;
+import gov.cdc.usds.simplereport.db.repository.OrganizationRepository;
+import gov.cdc.usds.simplereport.db.repository.PersonRepository;
+import gov.cdc.usds.simplereport.db.repository.ProviderRepository;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.test.context.ActiveProfiles;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * Created by nickrobison on 11/21/20
+ */
+@DataJpaTest
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+@ActiveProfiles("dev")
+class UploadServiceTest {
+
+    private final PersonService _ps;
+    private final UploadService _service;
+
+    public UploadServiceTest(@Autowired OrganizationRepository orgRepo, @Autowired ProviderRepository providerRepo, @Autowired PersonRepository repo) {
+        OrganizationService os = new OrganizationService(orgRepo, providerRepo);
+        this._ps = new PersonService(os, repo);
+        this._service = new UploadService(_ps);
+    }
+
+    @Test
+    void testInsert() throws IOException {
+        // Read the test CSV file
+        try (InputStream inputStream = UploadServiceTest.class.getClassLoader().getResourceAsStream("test-upload.csv")) {
+            this._service.processPersonCSV(inputStream);
+        }
+
+        final StreetAddress address = new StreetAddress("123 Main Street", "", "Washington", "DC", "20008", "");
+
+
+        final List<Person> patients = this._ps.getPatients();
+        assertAll(() -> assertEquals(1, patients.size()),
+                () -> assertEquals("Doe", patients.get(0).getFirstName()),
+                () -> assertEquals(address, patients.get(0).getAddress(), "Should have the correct address"));
+    }
+
+    @Test
+    void testNotCSV() throws IOException {
+        try (ByteArrayInputStream bis = new ByteArrayInputStream("this is not a CSV\n".getBytes(StandardCharsets.UTF_8))) {
+            this._service.processPersonCSV(bis);
```

Comment:
No exception here? I would think we'd want to return a 400 Bad Request in this case.

Reply:
Fixed! It's a bit hacky because the CSV parser doesn't throw an exception when given a single row, even if it's nonsense. So we throw one on our own.